File: AD-HOC/1987/1987.java
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		
		while(scan.hasNext()) {
			int qtdDigitos = scan.nextInt();
			char[] digitos = scan.next().toCharArray();
			int algarismosSomados = 0;
			
			for(int i=0; i<qtdDigitos; i++) {
				algarismosSomados += digitos[i] - 48;
			}
			if(algarismosSomados%3 == 0) System.out.println(algarismosSomados + " sim");
			else System.out.println(algarismosSomados + " nao");
		}
		scan.close();
	}
}

File: AD-HOC/1987/1987.cpp
#include <iostream>

using namespace std;

int main () {
	int qtdDigitos;
	string digitos;
	
	while(scanf("%d ", &qtdDigitos) != EOF) {
		getline(cin, digitos);
		int algarismosSomados = 0;
		
		for(int i=0; i<qtdDigitos; i++) {
			algarismosSomados += digitos[i] - 48;
		}
		if(algarismosSomados%3 == 0) printf("%d sim\n", algarismosSomados);
		else printf("%d nao\n", algarismosSomados);
	}
}
